SOILS

3 short skits
step.through ('soils2014')
step.through (demo="http://...")



position <- metadata (xx, "latitute|longitude")
library(spatstat)			# library(sp)
pp <- coords (position)
plot (pp)



<environment: namespace:utils>
> history()

 assign ("API", load.MGRAST('foo'), .MGRAST)
doc.MGRAST(5,head="matrix",str="source")
step.through('short')
step.through('short')
xx <- biom
xx <- biomRequest('mgp10307', request='organism', source='Greengenes')
t_xx <- transform(xx, t_Log)
image(t_xx)
columns (t_xx, "id")
columns (t_xx, "biom|temp"")
columns (t_xx, "biom|temp")
columns (t_xx, "biom|id")
image(t_xx, labCol = "$$project.id")
image(t_xx, labCol = "$$project.id", labRow="taxonomy1")
image(t_xx, labCol = "$$project.id", labRow="$$taxonomy2")
image(t_xx, labCol = "$$project.id", labRow="$$taxonomy2")
princomp(t_xx, map (col="biome"))
princomp(t_xx, map = c(col="biome"))
princomp(t_xx, map = c(col="biome"), labels="")
princomp(t_xx, map = c(col="biome"), labels="$$ncbi")
princomp(t_xx, map = c(col="biome"), labels="$$sample.id")
princomp(t_xx, map = c(pch="biome"))
princomp(t_xx, map = c(pch="biome"), cex=2)
dir.MGRAST()
history
history()








...continuation:

# projects can be analyzed together
# and metagenomes can be hand-picked (including from different projects)
also:
biom (file="xxxx.biom")
as.character (file="....")







biomRequest(...)
metadata(...)

offline demo

principal coord

using biom objects
dumping to file
input from qiime
